src/vast-ima-player.ts

```diff
@@ -1009,13 +1013,36 @@ export class Player extends DelegatedEventTarget {
     }
   }
 
+  private _onEnded() {
+    if (
+      this.isCustomPlaybackUsed() &&
+      this.#mediaElement.currentTime === this.#mediaElement.duration &&
+      this.#cuePoints.indexOf(-1) > -1
+    ) {
+      /* Fixing a bug with postroll on iOS
+       * when a postroll gets started via "contentComplete"
+       * and a single video-tag is used IMA does not reset
+       * back to the content after the postroll finished.
+       * Setting the time of the video tag a little shorter
+       * than the duration, so that the video-tag is not set to "ended".
+       */
+      this.#mediaElement.currentTime = this.#mediaElement.duration - 0.00001;
+      this.#customPlaybackTimeAdjustedOnEnded = true;
+    }
+    this.dispatchEvent(new CustomEvent('ended'));
+    this.#adsLoader.contentComplete();
+    if (!this.#adsManager) {
+      this._mediaStop();
+    }
+  }
+
   private _mediaStop() {
-    setTimeout(() => {
```

Comment:
I removed this setTimeout because `ended` now gets triggered manually instead of bubbling the event in `_handleMediaElementEvents`. Now it is ensured that `ended` comes first.